File: app_watcher/src/main/java/com/tongchen/twatcher/gank/ui/adapter/FragmentAdapter.java
package com.tongchen.twatcher.gank.ui.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.tongchen.twatcher.gank.model.entity.Category;

import java.util.List;

/**
 * Created by TongChen on 2018/9/30.
 * <p>
 * Description:
 */

public class FragmentAdapter extends FragmentPagerAdapter {

    private List<Fragment> mFragmentList;
    private List<Category> mCategoryList;

    public FragmentAdapter(FragmentManager fm) {
        super(fm);
    }

    public FragmentAdapter(FragmentManager fm, List<Fragment> fragments, List<Category> categories) {
        super(fm);
        mFragmentList = fragments;
        mCategoryList = categories;
    }

    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mCategoryList.get(position).getCategoryName();
    }
}

File: app_watcher/src/main/java/com/tongchen/twatcher/gank/ui/adapter/FragmentAdapter.kt
package com.tongchen.twatcher.gank.ui.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import com.tongchen.twatcher.gank.model.entity.Category

/**
 * @author TongChen
 * @date 2019/11/20  15:19
 * <p>
 * Desc:
 */
class FragmentAdapter : FragmentPagerAdapter {

    private lateinit var mFragmentList: MutableList<Fragment>
    private lateinit var mCategoryList: MutableList<Category>

    constructor(fragmentManager: FragmentManager) : super(fragmentManager) {

    }

    constructor(fragmentManager: FragmentManager, fragments: MutableList<Fragment>, categories: MutableList<Category>) : super(fragmentManager) {
        mFragmentList = fragments
        mCategoryList = categories
    }

    override fun getItem(position: Int): Fragment {
        return mFragmentList[position]
    }

    override fun getCount(): Int {
        return mFragmentList.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return mCategoryList[position].mCategoryName
    }
}